Creating a blank file causes a restore error

Attempting to initialise the config in the workdir by creating the file (`config.yaml`) raises an error, probably because of the "malformed" file.

```python
    def read(self, attr: str) -> Any:
        """Read the config file and return its contents.
    
        If it does not exist, creates the file and fills it with default vaulues.
        """
        if not self.path.exists():
            self.write()
        data = self._read()
>       if attr in data:
           ^^^^^^^^^^^^
E       TypeError: argument of type 'NoneType' is not iterable

/home/ljbeal/Work/Devel/ConfigLite/configlite/config.py:106: TypeError
```

Weirdly, it looks like the yaml safe_load is returning None if it can't load the file?

--- a/configlite/config.py
+++ b/configlite/config.py
@@ -1,5 +1,6 @@
 import os
 from pathlib import Path
+import shutil
 from typing import Any
 import yaml
 
@@ -94,8 +95,17 @@ class BaseConfig:
         self._ensure_dir()
         with self.path.open("r") as f:
             data = yaml.safe_load(f)
-        if data is not None:
+        if isinstance(data, dict):
             return data
+        backup_name = f"{self.filename}.bk"
+        print(f"WARNING: Config file {self.path} failed to load. Backing up the file to: {backup_name}...", end = " ")
+        try:
+            shutil.move(self.abspath, backup_name)
+        except:
+            print("Error.")
+            raise
+        else:
+            print("Done.")
         return {}
 
     def read(self, attr: str) -> Any:

--- a/tests/test_healing.py
+++ b/tests/test_healing.py
@@ -1,3 +1,4 @@
+import os
 from pathlib import Path
 from typing import Any
 
@@ -17,7 +18,7 @@ def verify_variable(file: Path, name: str, value: Any) -> bool:
     return data.get(name, None) == value
 
 
-def test_restore_file():
+def test_restore_file(capsys):
     file = Path("test.yaml")
 
     config = ConfigTest(file)
@@ -25,9 +26,26 @@ def test_restore_file():
     with file.open("w+") as o:
         o.write("")  # create an empty file
 
-    
     assert config.foo == "foo"
     assert config.val == 10
+
+    assert os.path.exists(f"{file}.bk")
+    assert "WARNING" in capsys.readouterr().out
+
+
+def test_mangled_file(capsys):
+    file = Path("test.yaml")
+
+    config = ConfigTest(file)
+
+    with file.open("w+") as o:
+        o.write("foo")  # create a broken file
+
+    assert config.foo == "foo"
+    assert config.val == 10
+
+    assert os.path.exists(f"{file}.bk")
+    assert "WARNING" in capsys.readouterr().out
 
 
 def test_delete_variable():

